### ISSUE TITLE
Changing any property on User extended model changes the emailVerified property to false

### ISSUE THREAD

**Issue by qaiser110** [strongloop/loopback#3063]
### Bug or feature request
- [x] Bug
- [ ] Feature request

### Issue Description 
I've I've recently migrated to Loopback 3. I'm Postgresql database. One of my tests is failing with a strange behaviour. I've raised [this issue](http://stackoverflow.com/q/41435187/699091) on Stackoverflow as well.

I have a Member model that extends the User model. Whenever I change any property on my Member model, the emailVerified changes from true to false

```js
Member.findById(memberId, function (err, user) {
  console.log(user.emailVerified);   // true

  user.updateAttribute('username', 'abc', function (err, result) {
    console.log(result.emailVerified);   // false
  })
});
```

I've also tried with updateAttributes but the results are the same.

I found that in user model code, emailChanged is returning true because ctx.data.email is undefined. Changing it to first check if it exists does solve my problem.

```js
var emailChanged = ctx.hookState.originalUserData.some(function(data) {
  return ctx.data.email && data.email != ctx.data.email;
});
```
### Additional information

Node v 7, 
LoopBack v 3.0

**Comment by mackwan84** [comment]
+1 I am facing the same problem after update to 3.0.

**Comment by hussainak** [comment]
+1

**Comment by notbrain** [comment]
+1 in 2.36.2

Our password reset is now broken because of this too. `emailVerified` set to 0.

Just tried to upgrade from our pinned 2.29.1 version of loopback to the latest 2.x version. Was pinned because of a previous regression wrt PUT/PATCH endpoint verbs. Trying to get back up to latest.

Tried to explicitly set `emailVerified = 1` in our custom reset function we created, but no dice (must be being reset in some "after" middleware in User? We extend with Person). 

Loopback 2.36.2 is not backwards compatible with 2.29.1. Sigh.

**Comment by lleopardi** [comment]
+1

**Comment by lschricke** [comment]
Same issue here.

**Comment by davidcheung** [comment]
once this lands it should fix 
https://github.com/strongloop/loopback/pull/3053

**Comment by davidcheung** [comment]
this bug should be fixed in #3053 or in (2.x) #3072, please ping me in this thread if issue persists to reopen. Closing now.
